### DISCUSSION TITLE
Help removing inherited members and extension methods sections

### DISCUSSION THREAD

**Discussion by lawsonmcw** [dotnet/docfx#9645] [Q&A] [answered]
Can anyone point me in the right direction as to what section of the class.tmpl.partial or class.header.tmpl.partial file(s) need to be edited in a custom template to completely remove the inheritance and extension methods sections? Every edit I have tried seems to have adverse side-effects on the other sections. I'm not familiar with web programming; I thought I could figure it out from trial and error, but I'm stuck. I've yet to find a concrete example of how to how to edit these from numerous searches. Thank you!

![image](https://github.com/dotnet/docfx/assets/1609488/1dcf0b50-e1ee-4053-9f64-29f4b1825045)

**Comment by BattlefieldDuck** [comment]
There are multiple solutions available that you might find helpful!

Adding a `filterConfig.yml`
See: https://github.com/dotnet/docfx/issues/1780

Overwriting the template or edit `_site\styles\main.css`
See: https://stackoverflow.com/questions/56458435/docfx-how-to-suppress-certain-info-about-type-inheritance-constructors-assem

**Comment by lawsonmcw** [comment]
Hi @BattlefieldDuck. Thank you for the reply! Yes, that was one of the first posts I found.

Starting with the simplest solution first, I added the suggested line ".inheritedMembers { display: none; }"  to my _site\styles\main.css file which had been empty. If I rebuild the site and reload the browser, nothing changes for me.

So I had been trying to edit the `class.header.tmpl.partial` file as suggested. Here's where I can't quite figure it out.  If I remove the following section starting at line 28 which seems to be roughly the right spot, it does remove the "Inherited Members" and "Extension Methods" sections.

```
{{#inheritedMembers.0}}
<div class="inheritedMembers">
  <h5>{{__global.inheritedMembers}}</h5>
{{/inheritedMembers.0}}
{{#inheritedMembers}}
  <div>
  {{#definition}}
    <xref uid="{{definition}}" text="{{nameWithType.0.value}}" alt="{{fullName.0.value}}"/>
  {{/definition}}
  {{^definition}}
    <xref uid="{{uid}}" text="{{nameWithType.0.value}}" alt="{{fullName.0.value}}"/>
  {{/definition}}
  </div>
{{/inheritedMembers}}
{{#inheritedMembers.0}}
</div>
{{/inheritedMembers.0}}
```
However, it also mangles the inheritance and implements sections above which had displayed so beautifully. So, this is where I'm stuck. I've tried many different edits to my custom `class.header.tmpl.partial` file (based on the default template) but I can't get it right. Any ideas on how I can edit this partial file correctly?

![image](https://github.com/dotnet/docfx/assets/1609488/18b51920-47c4-4ea4-b01e-ebd27541fb6d)

By the way, my template inheritance looks like this:
```
    "template": [
      "default",
      "modern",
      "templates/myCustomTemplate"
    ],
```

**Comment by BattlefieldDuck** [comment, marked as answer]
This is my own solution that I want to share.

1. Create a CSS file in the `docs/template/public/main.css` directory. Add the following CSS rules to this file:
```css
/* This rule hides the display of inherited types */
.typelist.inheritance {
    display: none;
}

/* This rule hides the display of implemented types */
.typelist.implements {
    display: none;
}

/* This rule hides the display of derived types */
.typelist.derived {
    display: none;
}
```

2. Modify the `docfx.json` File. Update the template section as follows:

```json
"template": [
  "default",
  "modern",
  "template"
],
```

After these changes, your documentation should reflect the new styling rules. Remember to rebuild your documentation for the changes to take effect.

**Comment by lawsonmcw** [comment]
Thank you @BattlefieldDuck !!  This was exactly the hint I needed to get going in the right direction.  Based on your instructions I created a main.css file within my customTemplate/public directory.  By playing around with it, I was able to eliminate the inherited members. It's not perfect in that I lost the derived types section which was useful, but it's a tradeoff worth making to get rid of dozens and dozens of inherited members in some classes. I was able to eliminate the extension methods section as well.  I would have guessed that .typelist.inheritedMembers would have done what I wanted, but no such luck (again, I know nothing about web programming so I'm taking wild stabs in the dark here.) Thanks for your help. I hope this is useful for others too.

```
/* This rule hides the display of inherited types */
.typelist.inheritance {
    display: contents;
}

/* This rule hides the display of implemented types */
.typelist.implements {
    display: contents;
}

/* This rule hides the display of derived types */
.typelist.derived {
    display: none;
}

/* this doesn't seem to work unfortunately
.typelist.inheritedMembers {
    display: none;
} /*

/* but this does! */
.typelist.extensionMethods {
    display: none;
} 

```
Much better:
![image](https://github.com/dotnet/docfx/assets/1609488/68090d7c-5850-4ccf-bd87-37405ab07be6)

**Reply by BattlefieldDuck** [reply]
I’m glad to hear that it works.

**Reply by BattlefieldDuck** [reply]
For the issue with `.typelist.inheritedMembers` not working, I have proposed a fix at https://github.com/dotnet/docfx/pull/9648.
